src/app/services/xml-parsers/apparatus-entries-parser.service.ts

```diff
@@ -0,0 +1,153 @@
+
+  constructor(
+    private editionDataService: EditionDataService,
+    private genericParserService: GenericParserService,
+    private witParserService: WitnessesParserService,
+  ) {
+  }
+
+  private parseAppEntriesList(document: XMLElement, witsNumber: number): ApparatusEntriesList {
+    const appEntries = Array.from(document.querySelectorAll<XMLElement>(this.appEntryTagName));
+
+    return this.parseAppEntries(appEntries, witsNumber);
+  }
+
+  private parseAppEntries(appEntries: XMLElement[], witsNumber: number) {
+    const parsedEntries = appEntries.map((appEntry) => this.parseAppEntry(appEntry, witsNumber));
+
+    return arrayToMap(parsedEntries, 'id');
+  }
+
+  private parseAppEntry(appEntry: XMLElement, witsNumber: number): ApparatusEntry {
+    const content = this.parseAppReadings(appEntry);
+
+    return {
+      type: 'ApparatusEntryComponent',
+      id: appEntry.getAttribute('xml:id') || xpath(appEntry),
+      attributes: this.genericParserService.parseAttributes(appEntry),
+      content,
+      notes: this.parseAppNotes(appEntry),
+      variance: this.calcVariance(content, witsNumber),
+      originalEncoding: getOuterHTML(appEntry),
+    };
+  }
+
+  private parseAppReadings(appEntry: XMLElement): Array<Lemma | Reading> {
+    return Array.from(appEntry.querySelectorAll(`${this.lemmaTagName}, ${this.readingTagName}`))
+      .map((rdg: XMLElement) => {
+        return rdg.tagName === this.lemmaTagName ? this.parseLemma(rdg) : this.parseReading(rdg);
+      });
+  }
+
+  private parseLemma(rdg: XMLElement): Lemma {
+    return {
+      type: 'LemmaComponent',
+      id: rdg.getAttribute('xml:id') || xpath(rdg),
+      attributes: this.genericParserService.parseAttributes(rdg),
+      witIDs: this.parseReadingWitnesses(rdg) || [],
+      content: this.parseAppReadingContent(rdg),
+      significant: true,
+    };
+  }
+
+  private parseReading(rdg: XMLElement): Reading {
+    return {
+      type: 'ReadingComponent',
+      id: rdg.getAttribute('xml:id') || xpath(rdg),
+      attributes: this.genericParserService.parseAttributes(rdg),
+      witIDs: this.parseReadingWitnesses(rdg) || [],
+      content: this.parseAppReadingContent(rdg),
+      significant: this.parseReadingSignificance(rdg),
+    };
+  }
+
+  private parseReadingWitnesses(rdg: XMLElement) {
+    return rdg.getAttribute('wit')?.split('#')
+      .map((el) => removeSpaces(el))
+      .filter((el) => el.length !== 0);
+  }
+
+  private parseAppReadingContent(rdg: XMLElement) {
+    return Array.from(rdg.childNodes)
+      .map((child: XMLElement) => {
+        if (child.nodeName === this.appEntryTagName) {
+          return {
+            type: 'ApparatusEntryComponent',
+            id: child.getAttribute('xml:id') || xpath(child),
+            attributes: {},
+            content: [],
+          };
+        }
+
+        return this.genericParserService.parse(child);
+      });
+  }
+
+  private parseReadingSignificance(rdg: XMLElement) {
```

Comment:
This function returns a boolean,  is it still a "parse" function?
Would an alternative name like `readingIsSignificant` fit better?